src/main/java/com/commercetools/pingpong/model/Game.java

```diff
@@ -9,78 +10,112 @@ public Game(Player firstPlayer, Player secondPlayer) {
         this.playerTwo = secondPlayer;
     }
 
-    public boolean hasSomebodyWon() {
-        return (hasPlayerWonSet(playerOne, playerTwo) || hasPlayerWonSet(playerTwo, playerOne));
+    public void updateScoreOfPlayer(Player player, int actionType) {
+
+        if (actionType == 1 && !hasSomebodyWonSet()) {
+            player.addPoint();
+            decideWhoServes();
+
+        } else if (actionType == 2 && player.getSetScore() > 0) {
+            player.subPoint();
+            whoServesAfterDoubleClick();
+
+        } else if (actionType == 3) {
+            resetSetScores();
+            resetMatchScores();
+            resetServe();
+
+        }
+
     }
 
-    public boolean hasOneOfThePlayersTwoPointsMore(Player playerWhoIsWinning, Player playerWhoIsLosing) {
-        return (playerWhoIsWinning.getSetScore() - playerWhoIsLosing.getSetScore() == 2);
+    public void updateMatchScoreOfPlayers() {
+        if (hasPlayerWonSet(playerOne, playerTwo) && !isMatchOver()) {
+            this.playerOne.addMatchScore();
+            resetSetScores();
+            setBeginningServe();
+
+        } else if (hasPlayerWonSet(playerTwo, playerOne) && !isMatchOver()) {
+            this.playerTwo.addMatchScore();
+            resetSetScores();
+            setBeginningServe();
+        }
+    }
+
+    public void isGameOver() {
+        if (this.playerOne.getMatchScore() == 2) {
+            resetSetScores();
+            resetMatchScores();
+            resetServe();
+            resetBeginningServe();
+        } else if (this.playerTwo.getMatchScore() == 2) {
+            resetSetScores();
```

Comment:
We are calling the exact same methods in both cases. We can combine them in one condition.

Reply:
of course 👍 
d5238189bea2597a7861e1fb2e59018c39ddeda6